pgpainless-core/src/main/java/org/pgpainless/util/ArmorUtils.java

```diff
@@ -106,14 +108,36 @@ public static String toAsciiArmoredString(PGPPublicKeyRingCollection publicKeyRi
         return sb.toString();
     }
 
+    private static Tuple<Optional<String>, Integer> getPrimaryUserIdAndUserIdCount(PGPPublicKey publicKey) {
```

Comment:
I'm not really happy with this implementation, any suggestions? I've thought about reusing KeyRingInfo, but this would be a lot of overhead (performance?) just for printing out that information.

Reply:
Since armor headers are not authenticated anyways, its probably enough to just count the user-ids manually? Not sure if there is a more efficient way to determine the true primary user-id though.